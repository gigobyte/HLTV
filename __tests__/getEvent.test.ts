import HLTV from '../src/'

function sleep(ms: number): Promise<void> {
  return new Promise(resolve => setTimeout(resolve, ms))
}

test('getEvent', async () => {
  await sleep(1000)
  expect(await HLTV.getEvent({ id: 3452 })).toMatchSnapshot()
  expect(await HLTV.getEvent({ id: 3552 })).toMatchSnapshot()
  expect(await HLTV.getEvent({ id: 355 })).toMatchSnapshot()
  await sleep(3000) // Sleep to prevent accidental ban by hltv
  expect(await HLTV.getEvent({ id: 3005 })).toMatchSnapshot()
  expect(await HLTV.getEvent({ id: 4356 })).toMatchSnapshot()
  expect(await HLTV.getEvent({ id: 2870 })).toMatchSnapshot()
  await sleep(1000)
}, 30000)
