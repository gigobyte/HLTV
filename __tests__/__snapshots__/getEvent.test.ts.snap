// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEvent 1`] = `
Object {
  "dateEnd": 1513422000000,
  "dateStart": 1513422000000,
  "formats": Array [
    Object {
      "description": "Single-elimination BO3",
      "type": "Playoffs",
    },
  ],
  "id": 3452,
  "location": Object {
    "code": "SE",
    "name": "Sweden",
  },
  "mapPool": Array [
    "cch",
    "mrg",
    "inf",
    "nuke",
    "trn",
    "cbl",
    "ovp",
  ],
  "name": "Ownit 2017 Winter Finals",
  "prizeDistribution": Array [
    Object {
      "otherPrize": undefined,
      "place": "1st",
      "prize": "$4,100",
      "qualifiesFor": undefined,
      "team": Object {
        "id": 6497,
        "name": "TzatzikiKlubben",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "2nd",
      "prize": "$1,175",
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8660,
        "name": "low riders",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "3rd",
      "prize": "$575",
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8659,
        "name": "Kampners Advokatbyrå",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "4th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7908,
        "name": "PopcornPlokker$",
      },
    },
  ],
  "prizePool": "$5,850",
  "relatedEvents": Array [],
  "teams": Array [
    Object {
      "id": 8659,
      "name": "Kampners Advokatbyrå",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 7908,
      "name": "PopcornPlokker$",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 6497,
      "name": "TzatzikiKlubben",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8660,
      "name": "low riders",
      "reasonForParticipation": undefined,
    },
  ],
}
`;

exports[`getEvent 2`] = `
Object {
  "dateEnd": 1520506800000,
  "dateStart": 1518519600000,
  "formats": Array [
    Object {
      "description": "Round-robin BO2",
      "type": "Group stage",
    },
    Object {
      "description": "Single-elimination BO3",
      "type": "Playoffs",
    },
  ],
  "id": 3552,
  "location": Object {
    "code": "EU",
    "name": "Europe",
  },
  "mapPool": Array [
    "cch",
    "mrg",
    "inf",
    "nuke",
    "trn",
    "cbl",
    "ovp",
  ],
  "name": "CSesport.com Cup #3",
  "prizeDistribution": Array [
    Object {
      "otherPrize": undefined,
      "place": "1st",
      "prize": "$2,000",
      "qualifiesFor": undefined,
      "team": Object {
        "id": 6343,
        "name": "EURONICS",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "2nd",
      "prize": "$500",
      "qualifiesFor": undefined,
      "team": Object {
        "id": 6652,
        "name": "GameAgents",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "3-4th",
      "prize": "$250",
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8361,
        "name": "SEAL",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "3-4th",
      "prize": "$250",
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7187,
        "name": "Nexus",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-6th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8572,
        "name": "Codewise Unicorns",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-6th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7017,
        "name": "Space Jam",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "7-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 4929,
        "name": "Refuse",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "7-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8305,
        "name": "DreamEaters",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "9-10th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 6679,
        "name": "Revolution",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "9-10th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8209,
        "name": "Pompa",
      },
    },
  ],
  "prizePool": "$3,000",
  "relatedEvents": Array [
    Object {
      "id": 3457,
      "name": "CSesport.com XMAS Cup",
    },
    Object {
      "id": 3638,
      "name": "CSesport.com Cup #4",
    },
  ],
  "teams": Array [
    Object {
      "id": 8572,
      "name": "Codewise Unicorns",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8305,
      "name": "DreamEaters",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 6343,
      "name": "EURONICS",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 6652,
      "name": "GameAgents",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 7187,
      "name": "Nexus",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8209,
      "name": "Pompa",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 4929,
      "name": "Refuse",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 6679,
      "name": "Revolution",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8361,
      "name": "SEAL",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 7017,
      "name": "Space Jam",
      "reasonForParticipation": undefined,
    },
  ],
}
`;

exports[`getEvent 3`] = `
Object {
  "dateEnd": undefined,
  "dateStart": undefined,
  "formats": Array [],
  "id": 355,
  "location": Object {
    "code": "WORLD",
    "name": "World",
  },
  "mapPool": Array [],
  "name": "DH 09 Summer Qualifier Copenhagen",
  "prizeDistribution": Array [],
  "prizePool": "TBA",
  "relatedEvents": Array [],
  "teams": Array [
    Object {
      "id": 334,
      "name": "PlayZeek",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 166,
      "name": "Spirit of Amiga",
      "reasonForParticipation": undefined,
    },
  ],
}
`;

exports[`getEvent 4`] = `
Object {
  "dateEnd": 1500976800000,
  "dateStart": 1500285600000,
  "formats": Array [],
  "id": 3005,
  "location": Object {
    "code": "AR",
    "name": "Argentina",
  },
  "mapPool": Array [],
  "name": "Liga TyC Sports Aorus #1",
  "prizeDistribution": Array [
    Object {
      "otherPrize": "Liga TyC Sports Aorus Finals",
      "place": "1-2nd",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 6799,
        "name": "Furious",
      },
    },
    Object {
      "otherPrize": "Liga TyC Sports Aorus Finals",
      "place": "1-2nd",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7653,
        "name": "Isurus",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "3-4th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8123,
        "name": "Keymotion",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "3-4th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8125,
        "name": "Hafnet",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8094,
        "name": "Licht",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8121,
        "name": "Malvinas",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8122,
        "name": "Kitsune",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 8124,
        "name": "Keymotion Red",
      },
    },
  ],
  "prizePool": "TBA",
  "relatedEvents": Array [],
  "teams": Array [
    Object {
      "id": 6799,
      "name": "Furious",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8125,
      "name": "Hafnet",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 7653,
      "name": "Isurus",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8123,
      "name": "Keymotion",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8124,
      "name": "Keymotion Red",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8122,
      "name": "Kitsune",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8094,
      "name": "Licht",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 8121,
      "name": "Malvinas",
      "reasonForParticipation": undefined,
    },
  ],
}
`;

exports[`getEvent 5`] = `
Object {
  "dateEnd": 1549623600000,
  "dateStart": 1549450800000,
  "formats": Array [
    Object {
      "description": "Double-Elimination BO3 BO5 Grand Final",
      "type": "Playoffs",
    },
  ],
  "id": 4356,
  "location": Object {
    "code": "NAM",
    "name": "North America",
  },
  "mapPool": Array [
    "cch",
    "d2",
    "mrg",
    "inf",
    "nuke",
    "trn",
    "ovp",
  ],
  "name": "IEM Sydney 2019 North America Closed Qualifier",
  "prizeDistribution": Array [
    Object {
      "otherPrize": undefined,
      "place": "1st",
      "prize": undefined,
      "qualifiesFor": Object {
        "id": 4236,
        "name": "IEM Sydney 2019",
      },
      "team": Object {
        "id": 7726,
        "name": "Swole Patrol",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "2nd",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7106,
        "name": "eUnited",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "3rd",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7801,
        "name": "Ghost",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "4th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 6290,
        "name": "Luminosity",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-6th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 5005,
        "name": "compLexity",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "5-6th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 5991,
        "name": "Envy",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "7-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 9799,
        "name": "Bad News Bears",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "7-8th",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 9804,
        "name": "Win Scrims Not Matches",
      },
    },
  ],
  "prizePool": "1 spot at IEM Sydney",
  "relatedEvents": Array [
    Object {
      "id": 4236,
      "name": "IEM Sydney 2019",
    },
    Object {
      "id": 4354,
      "name": "IEM Sydney 2019 North America Open Qualifier 1",
    },
    Object {
      "id": 4355,
      "name": "IEM Sydney 2019 North America Open Qualifier 2",
    },
  ],
  "teams": Array [
    Object {
      "id": 5991,
      "name": "Envy",
      "reasonForParticipation": "Invited",
    },
    Object {
      "id": 7801,
      "name": "Ghost",
      "reasonForParticipation": "Invited",
    },
    Object {
      "id": 6290,
      "name": "Luminosity",
      "reasonForParticipation": "Invited",
    },
    Object {
      "id": 5005,
      "name": "compLexity",
      "reasonForParticipation": "Invited",
    },
    Object {
      "id": 9804,
      "name": "Win Scrims Not Matches",
      "reasonForParticipation": "Qualifier 1",
    },
    Object {
      "id": 7106,
      "name": "eUnited",
      "reasonForParticipation": "Qualifier 1",
    },
    Object {
      "id": 9799,
      "name": "Bad News Bears",
      "reasonForParticipation": "Qualifier 2",
    },
    Object {
      "id": 7726,
      "name": "Swole Patrol",
      "reasonForParticipation": "Qualifier 2",
    },
  ],
}
`;

exports[`getEvent 6`] = `
Object {
  "dateEnd": 1495360800000,
  "dateStart": 1495360800000,
  "formats": Array [],
  "id": 2870,
  "location": Object {
    "code": "SE",
    "name": "Sweden",
  },
  "mapPool": Array [
    "cch",
    "mrg",
    "inf",
    "nuke",
    "trn",
    "cbl",
    "ovp",
  ],
  "name": "GAMERZ Season 1: Nordic Edition",
  "prizeDistribution": Array [
    Object {
      "otherPrize": "Pro contracts",
      "place": "1st",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7930,
        "name": "GAMERZ BLUE",
      },
    },
    Object {
      "otherPrize": undefined,
      "place": "2nd",
      "prize": undefined,
      "qualifiesFor": undefined,
      "team": Object {
        "id": 7931,
        "name": "GAMERZ RED",
      },
    },
  ],
  "prizePool": "Professional contracts",
  "relatedEvents": Array [],
  "teams": Array [
    Object {
      "id": 7930,
      "name": "GAMERZ BLUE",
      "reasonForParticipation": undefined,
    },
    Object {
      "id": 7931,
      "name": "GAMERZ RED",
      "reasonForParticipation": undefined,
    },
  ],
}
`;
